zenoh/src/session.rs

```diff
@@ -948,7 +987,15 @@ impl Session {
             callback,
         });
 
-        let declared_sub = (origin != Locality::SessionLocal)
+        #[cfg(not(feature = "unstable"))]
+        let declared_sub = origin != Locality::SessionLocal;
```

Comment:
Locality should be an unstable feature. So both cases should be under the "unstable" feature gate.

Reply:
Locality is public when unstable feature is enabled and private otherwise.